src/inmanta/server/services/environmentservice.py

```diff
@@ -519,13 +531,27 @@ async def environment_create_token(self, env: data.Environment, client_types: li
 
     @handle(methods_v2.environment_settings_list, env="tid")
     async def environment_settings_list(self, env: data.Environment) -> model.EnvironmentSettingsReponse:
+        settings: dict[str, model.EnvSettingType] = env.settings.get_all_setting_values()
         return model.EnvironmentSettingsReponse(
-            settings=dict(sorted(env.settings.items())),
+            settings=dict(sorted(settings.items())),
+            settings_v2={
+                setting_name: model.EnvironmentSettingDetails(
+                    value=v.value,
+                    protected=v.protected,
+                    protected_by=v.protected_by,
+                )
+                for setting_name, v in sorted(env.settings.settings.items())
+            },
```

Comment:
Isn't this equivalent to the following, or am I missing something?
```suggestion
            settings_v2=dict(sorted(env.settings.settings.items())),
```